tools/check-upcoming-maintenance.py

```diff
@@ -0,0 +1,130 @@
+import os
+import re
+import subprocess
+from typing import List
+from optparse import OptionParser
+
+class NodeMaintenance:
+    '''
+    Class to keep track of project, zones, and search terms for nodes, as well as results from gcloud queries
+    '''
+    def __init__(self, project, zones, regex):
+        self.project = project
+        res = subprocess.run("gcloud projects describe {}".format(self.project), shell=True, capture_output=True, text=True)
+        if res.returncode != 0:
+            raise Exception("{} does not exist or you may not have permission to access it".format(self.project))
+        
+        self.zones = None
+        if zones:
+            self.zones = zones.split(",")
+            for zone in self.zones:
+                res = subprocess.run("gcloud compute zones describe {}".format(zone), shell=True, capture_output=True, text=True)
+                if res.returncode != 0:
+                    raise Exception("{} does not exist".format(zone))
+        
+        self.regex = None
+        self.screen_reader = False
```

Comment:
Most gcloud commands take [format](https://cloud.google.com/sdk/gcloud/reference#--format) argument. I would suspect this is to aid automation. I think picking a format like yaml or json would avoid the screen reader issue and would be more robust in the long term.